bindings/python/mainloop.pyx

```diff
@@ -0,0 +1,17 @@
+
+cdef extern from "sol-mainloop.h":
+	int sol_init()
+	int sol_run()
+	void sol_shutdown()
+
+class Mainloop():
+	def __init__(self):
+		sol_init()
+
+	def __del__(self):
+		sol_shutdown()
```

Comment:
good. But now I'm thinking if we should  do these in class or automatically. Likely automatically would make more sense, since calling soletta methods before `init()` is not possible. In EFL we did it automatically, see: https://git.enlightenment.org/bindings/python/python-efl.git/tree/efl/ecore/efl.ecore.pyx

```py
def init():
    sol_init()

def shutdown():
    sol_shutdown()

init()
atexit.register(shutdown)
```

But you can hold on that. I need to think how to integrate with `asyncio`. One option is to have soletta to be a guest mainloop to that one, like I'm proposing to node.js. Another is to make `asyncio` use soletta.

My main concern with the second idea is that soletta itself will do `sigprocmask` and handle signals, that may conflict with python's own. So we'll need a way to cooperate both. If we can forward soletta ML primitives to `asyncio`, then we can, **before** calling `sol_init()` call `sol_mainloop_set_implementation()` with a set of `cdef` functions that wraps and forwards soletta to asyncio and back.